flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -378,6 +381,11 @@ public void notifyBufferDestroyed() {
 		// Nothing to do actually.
 	}
 
+	@VisibleForTesting
+	public void increaseCredit(int credit) {
```

Comment:
I considered this way before, but it seems more complicated to do that and exceeds the scope of this tests.

My initial idea is just to verify the logic related with handler, and the preceding logic related with `RemoteInputChannel` already covered in `RemoteInputChannelTest`. 

I know that it may be better to verify the whole process here. But it needs to use the real `PartitionRequestClient` to notify the credit via `onSenderBacklog`, and the current `PartitionRequestClient` is tied to `PartitionRequestClientHandler` not `CreditBasedClientHandler`.

Nevertheless, I will try to do that as your suggested way. :)

Reply:
or how about recycling a `Buffer` instance into the channel via `RemoteInputChannel#recycle()?  That would also be part of the current interface and yield the same result but is a bit more hacky (but similar to calling `onSenderBacklog` with the advantage being closer to your `increaseCredit` method).

Reply:
It is no problem to increase credit by either `BufferResponse` or `recycle` methods. 

The only concern is the way to trigger `notifyCreditAvailable`. I think it may still need to trigger by calling `CreditBasedClientHandler#notifyCreditAvailable` explicitly. The common trigger way seems not work currently via `RemoteInputChannle` -> `PartitionRequestClient` -> `PartitionRequestClientHandler` because the `PartitionRequestClient` is still tied to the old `PartitionRequestClientHandler`.

Reply:
We increase credit here only to confirm the `checkState` of credit correctly during sending `AddCredit` message, not to trigger `notifyCreditAvailable` by increasing credit.